File: 2025/424. Longest Repeating Character Replacement/Solution.java
class Solution {
    public int characterReplacement(String s, int k) {
        int[] freq = new int[26];
        int i = 0, j = 0;
        int max = 0, high = 0;
        char[] arr = s.toCharArray();
        while (j < arr.length) {
            freq[arr[j] - 'A']++;
            high = Math.max(high, freq[arr[j] - 'A']);
            if (j - i + 1 - high > k) {
                freq[arr[i++] - 'A']--;
            }
            max = Math.max(max, j - i + 1);
            j++;
        }
        return max;
    }
}

File: 2025/424. Longest Repeating Character Replacement/solution.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        map = {}
        i = j = freq = res = 0
        while j < len(s):
            map[s[j]] = map.get(s[j], 0) + 1
            freq = max(freq, map[s[j]])
            if (j - i + 1 - freq > k):
                map[s[i]] = map.get(s[i]) - 1
                i += 1
            res = max(res, j - i + 1)
            j += 1
        return res